flytekit/common/workflow_execution.py

```diff
@@ -36,9 +42,26 @@ def inputs(self):
         :rtype:  dict[Text, T]
         """
         if self._inputs is None:
-            self._inputs = _type_helpers.unpack_literal_map_to_sdk_python_std(
-                _engine_loader.get_engine().get_workflow_execution(self).get_inputs()
-            )
+            client = _flyte_engine._FlyteClientManager(
+                _platform_config.URL.get(), insecure=_platform_config.INSECURE.get()
+            ).client
+            execution_data = client.get_execution_data(self.id)
+
+            # Inputs are returned inline unless they are too big, in which case a url blob pointing to them is returned.
+            if bool(execution_data.full_inputs.literals):
+                return execution_data.full_inputs
+
+            if execution_data.inputs.bytes > 0:
+                with _common_utils.AutoDeletingTempDir() as t:
```

Comment:
I actually want to get rid of the concept of AutoDeletingTempDir in the framework code. I think this should be a user decision